homeassistant/util/logging.py

```diff
@@ -24,104 +27,51 @@ def filter(self, record: logging.LogRecord) -> bool:
-        self.flush = handler.flush
-        self.handle = handler.handle
-        self.handleError = handler.handleError
-        self.format = handler.format
-
-        self._thread.start()
-
-    def close(self) -> None:
-        """Wrap close to handler."""
-        self.emit(None)
-
-    async def async_close(self, blocking: bool = False) -> None:
-        """Close the handler.
-
-        When blocking=True, will wait till closed.
-        """
-        await self._queue.put(None)
-
-        if blocking:
-            while self._thread.is_alive():
-                await asyncio.sleep(0)
-
-    def emit(self, record: Optional[logging.LogRecord]) -> None:
-        """Process a record."""
-        ident = self.loop.__dict__.get("_thread_ident")
-
-        # inside eventloop
-        if ident is not None and ident == threading.get_ident():
-            self._queue.put_nowait(record)
-        # from a thread/executor
-        else:
-            self.loop.call_soon_threadsafe(self._queue.put_nowait, record)
-
-    def __repr__(self) -> str:
-        """Return the string names."""
-        return str(self.handler)
-
-    def _process(self) -> None:
-        """Process log in a thread."""
+class HomeAssistantQueueHandler(logging.handlers.QueueHandler):
+    """Process the log in another thread."""
+
+    def emit(self, record: logging.LogRecord) -> None:
+        """Emit a log record."""
         try:
-            while True:
-                record = asyncio.run_coroutine_threadsafe(
-                    self._queue.get(), self.loop
-                ).result()
-
-                if record is None:
-                    self.handler.close()
-                    return
-
-                self.handler.emit(record)
-        except asyncio.CancelledError:
-            self.handler.close()
-
-    def createLock(self) -> None:  # pylint: disable=invalid-name
-        """Ignore lock stuff."""
-
-    def acquire(self) -> None:
-        """Ignore lock stuff."""
-
-    def release(self) -> None:
-        """Ignore lock stuff."""
-
-    @property
-    def level(self) -> int:
-        """Wrap property level to handler."""
-        return self.handler.level
-
-    @property
-    def formatter(self) -> Optional[logging.Formatter]:
-        """Wrap property formatter to handler."""
-        return self.handler.formatter
-
-    @property
-    def name(self) -> str:
-        """Wrap property set_name to handler."""
-        return self.handler.get_name()  # type: ignore
-
-    @name.setter
-    def name(self, name: str) -> None:
-        """Wrap property get_name to handler."""
-        self.handler.set_name(name)  # type: ignore
+            self.enqueue(record)
+        except asyncio.CancelledError:  # pylint: disable=try-except-raise
+            raise
+        except Exception:  # pylint: disable=broad-except
+            self.handleError(record)
+
+
+@callback
+def async_migrate_log_handlers_to_queue(hass: HomeAssistant) -> None:
```

Comment:
I think this is not a migration since there's no persistent data that we need to migrate. We're just activating this log queue handler. I'd call this `async_activate_log_queue_handler` or `async_start_log_queue_handler`.

Reply:
`async_activate_log_queue_handler` Makes the most sense to me so I went with that